### ISSUE TITLE
0.5.0 breaks artifactId in dependencies for local modules

### ISSUE THREAD

**Issue by gabrielittner** [vanniktech/gradle-maven-publish-plugin#18]
I'm using your project as an example https://github.com/gabrielittner/Emoji. For that I've changed the artifactId of the `emoji` module to `emoji-base` but kept the module name itself.  

This is what it looks like: 
- https://github.com/gabrielittner/Emoji/commit/f5550109b1a05589fdea4be4556b816da0ea4aaa
- [emoji-base pom](https://oss.sonatype.org/content/repositories/snapshots/com/gabrielittner/emojitest/emoji-base/test1-SNAPSHOT/emoji-base-test1-20180823.130554-1.pom)
- [emoji-google pom](https://oss.sonatype.org/content/repositories/snapshots/com/gabrielittner/emojitest/emoji-google/test1-SNAPSHOT/emoji-google-test1-20180823.130612-1.pom) (correctly depends on the changed artifactId)

After upgrading to 0.5.0 it looks like this:
- https://github.com/gabrielittner/Emoji/commit/b1a7a9e54481c79b590cdd214ab973821d4e8376
- [emoji-base pom](https://oss.sonatype.org/content/repositories/snapshots/com/gabrielittner/emojitest/emoji-base/test2-SNAPSHOT/emoji-base-test2-20180823.131255-1.pom)
- [emoji-google pom](https://oss.sonatype.org/content/repositories/snapshots/com/gabrielittner/emojitest/emoji-google/test2-SNAPSHOT/emoji-google-test2-20180823.131314-1.pom)

Now `emoji-google` and the other artifacts don't have `emoji-base` as artifactId in their dependency but the module name. Instead they depend on the non-existent `emoji` artifact (the module name).

**Comment by vanniktech** [comment]
Which task have you used for that? Is it the one I added in https://github.com/vanniktech/gradle-maven-publish-plugin/pull/17?

**Comment by gabrielittner** [comment]
I've used the regular `uploadArchives`

**Comment by vanniktech** [comment]
The only thing that changed between 0.4.0 and 0.5.0 is the new task.

https://github.com/vanniktech/gradle-maven-publish-plugin/compare/0.4.0...master

I doubt that it changed anything.

**Comment by vanniktech** [comment]
Sometimes the upload with `uploadArchives` does not really work on projects where you execute the task in parallel or with a Daemon. At least that's my experience.

I release the Emoji library via `./gradlew uploadArchives --no-daemon --no-parallel`

**Comment by gabrielittner** [comment]
I'm doing the same. It also happens for snapshots published from my CI, so it shouldn't be anything local on my machine.

**Comment by vanniktech** [comment]
Could it be that in the meantime a new Gradle version was used and the Maven Plugin itself changed? Otherwise I can't really explain that behavior since the task itself is untouched.

**Comment by gabrielittner** [comment]
I don't think so. Both of the two linked commits above use Gradle 4.8. The project that I've noticed it on used Gradle 4.9 and the behavior is the same there (works together with 0.4.0, breaks with 0.5.0).

**Comment by gabrielittner** [comment]
Adding `pom.artifactId = project.findProperty("POM_ARTIFACT_ID")` to the mavenDeployer block of installArchives (line 162) fixes uploadArchives.

**Comment by vanniktech** [comment]
I guess the mavenDeployer should be reused then. Since it uses much more than just the artifact id

**Comment by vanniktech** [comment]
I still don't understand why adding something for the `installArchives` task has some effect on `uploadArchives`. Do you know the reason?

**Comment by gabrielittner** [comment]
I didn't have enough time to look into that yet

**Comment by gabrielittner** [comment]
I was able to track it down. This is the code that determines the artifactId of local modules/projects for the dependency list in the pom https://github.com/gradle/gradle/blob/e975872ae2cf1d521b73f1107cae772ca09c63cd/subprojects/maven/src/main/java/org/gradle/api/publication/maven/internal/pom/ProjectDependencyArtifactIdExtractorHack.java#L60-L68

Every module/project has two Upload tasks. The linked code gets those, looks at the poms of their deployers and extracts the artifactId. The uploadArchives deployer sets the id to `POM_ARTIFACT_ID`, while installArchives doesn't set anything which means that it uses the module/project name by default. Now the extractor has two different artifactIds and it falls back to using either archivesBaseName or the module/project name.

I'll open a PR to fix it.
